GalaxyZooTouchTable/GalaxyZooTouchTable/ViewModels/LevelerViewModel.cs

```diff
@@ -0,0 +1,124 @@
+﻿using GalaxyZooTouchTable.Lib;
+using GalaxyZooTouchTable.Models;
+using GalaxyZooTouchTable.Utility;
+using System.ComponentModel;
+using System.Windows.Input;
+
+namespace GalaxyZooTouchTable.ViewModels
+{
+    public class LevelerViewModel : INotifyPropertyChanged
+    {
+        public TableUser User { get; set; }
+        public ICommand ToggleLeveler { get; set; }
+
+        private int _classificationsUntilUpgrade { get; set; } = 6;
+        public int ClassificationsUntilUpgrade
+        {
+            get { return _classificationsUntilUpgrade; }
+            set
+            {
+                if (value <= 0)
+                {
+                    value = 6;
+                    LevelUp();
+                }
+                _classificationsUntilUpgrade = value;
+                OnPropertyRaised("ClassificationsUntilUpgrade");
+            }
+        }
+
+        private int _classificationsThisSession { get; set; } = 0;
+        public int ClassificationsThisSession
+        {
+            get { return _classificationsThisSession; }
+            set
+            {
+                ClassificationsUntilUpgrade--;
+                _classificationsThisSession = value;
+                OnPropertyRaised("ClassificationsThisSession");
+            }
+        }
+
+        private string _classificationLevel { get; set; } = "One";
+        public string ClassificationLevel
+        {
+            get { return _classificationLevel; }
+            set
+            {
+                _classificationLevel = value;
+                OnPropertyRaised("ClassificationLevel");
+            }
+        }
+
+        private bool _levelerIsOpen = false;
+        public bool LevelerIsOpen
+        {
+            get { return _levelerIsOpen; }
+            set
+            {
+                _levelerIsOpen = value;
+                OnPropertyRaised("LevelerIsOpen");
+            }
+        }
+
+        public LevelerViewModel(TableUser user = null)
+        {
+            User = user;
+            LoadCommands();
+
+            if (user != null)
+            {
+                Messenger.Default.Register<int>(this, OnClassificationReceived, user);
+            }
+        }
+
+        private void OnClassificationReceived(int TotalClassifications)
+        {
+            ClassificationsThisSession = TotalClassifications;
+        }
+
+        public void LoadCommands()
+        {
+            ToggleLeveler = new CustomCommand(SlideLeveler);
+        }
+
+        private void SlideLeveler(object sender)
+        {
+            LevelerIsOpen = !LevelerIsOpen;
+        }
+
+        private void LevelUp()
+        {
+            if (ClassificationsThisSession > 25)
```

Comment:
Not sure why you have this here. Wouldn't the switch just resolve to "one" on values greater than 25? Isn't that what you want?

Reply:
Not quite. There's a difference between "leveling up" and advancing by submitting a classification. I think this will be more apparent if you open this branch locally. The `LevelUp` method is responsible for advancing from level one to level five (the max level), and a user advances to a new level once six classifications have been submitted.

However, for readability I might be able to insert a constant here that reflects the "max level" has been reached and the method should be escaped. As of now, level five is the max level, but that could change in the future.